component/src/main/java/org/wso2/extension/siddhi/io/rabbitmq/source/RabbitMQSource.java

```diff
@@ -379,18 +381,23 @@ protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
     public void disconnect() {
         if (connection != null) {
             try {
-                RabbitMQConsumer.closeChannel();
-                connection.close();
-                if (log.isDebugEnabled()) {
-                    log.debug("Server connector for uri = " + listenerUri + " is disconnected in " +
-                            "" + sourceEventListener + ".");
+                if (connection.isOpen()) {
+                    rabbitMQConsumer.closeChannel();
+                    connection.close();
+                    if (log.isDebugEnabled()) {
+                        log.debug("Server connector for uri = " + listenerUri + " is disconnected in " +
+                                "" + sourceEventListener + ".");
+                    }
                 }
             } catch (TimeoutException e) {
                 log.error("Timeout while disconnecting the uri = " + listenerUri + " in " +
                         "" + sourceEventListener + ".");
             } catch (IOException e) {
                 log.error("Error in disconnecting the uri = " + listenerUri + " in " +
                         "" + sourceEventListener + ".");
+            } catch (Exception e) {
+                log.error("Error occurred while closing the RabbitMQ consumer for the queue: "
```

Comment:
Add the Siddhi app name and the associated stream name to the logs

Reply:
Added more context  details for the exception log.